pkg/migrator/cleanup.go

```diff
@@ -0,0 +1,246 @@
+	if err != nil {
+		return nil, err
+	}
+	return datastoreCs, nil
+}
+
+// findCsFromOpsrc extracts the packageName from a Subscription,
+// and finds the corresponding CatalogSource that the package belongs to.
+func findCsFromOpsrc(subscription *olm.Subscription, kubeClient client.Client) (*olm.CatalogSource, error) {
+	packageName := getSubsPackageName(subscription)
+	opsrcs := &v1.OperatorSourceList{}
+	err := kubeClient.List(context.TODO(), &client.ListOptions{}, opsrcs)
+	if err != nil {
+		return nil, err
+	}
+	for _, instance := range opsrcs.Items {
+		if !isPackageInOpsrc(packageName, &instance) {
+			continue
+		}
+		// fetch the CatalogSource with the same name
+		datastoreCs := &olm.CatalogSource{}
+		namespacedName := types.NamespacedName{Name: instance.GetName(), Namespace: instance.GetNamespace()}
+		err = kubeClient.Get(context.TODO(), namespacedName, datastoreCs)
+		if err != nil {
+			return nil, err
+		}
+		return datastoreCs, nil
+	}
+	return nil, nil
+}
+
+// isPackageInOpsrc takes the name of a package, and an OperatorSource
+// as input, and returns true if the package is present in the OperatorSource's
+// status.Packages field
+func isPackageInOpsrc(packageName string, opsrc *v1.OperatorSource) bool {
+	packages := getOpsrcPackages(opsrc)
+	for _, pkg := range packages {
+		if pkg == packageName {
+			return true
+		}
+	}
+	return false
+}
+
+// extractCsName takes the name of a CatalogSourceConfig
+// as input, and infers the name of the corresponding
+// CatalogSource datastore. Installed CSCs follow the
+// following naming pattern: installed-publisher-namespace.
+// For example, for a CatalogSourceConfig named
+// `installed-community-openshift-operators`,
+// extractCsName returns `community-operators` as output
+func extractCsName(cscName string) (string, error) {
+	var stringBuilder strings.Builder
+	possibleCsName := strings.Split(cscName, "-")[1]
+	_, err := stringBuilder.WriteString(possibleCsName)
+	if err != nil {
+		return "", err
+	}
+	_, err = stringBuilder.WriteString("-operators")
+	if err != nil {
+		return "", err
+	}
+	return stringBuilder.String(), nil
+}
+
+// getOpsrcPackages returns the list of packages from the Status block
+func getOpsrcPackages(opsrc *v1.OperatorSource) []string {
+	return strings.Split(opsrc.Status.Packages, ",")
+}
+
+// getSubsPackageName returns the packageName of a Subscription
+func getSubsPackageName(subscription *olm.Subscription) string {
+	return subscription.Spec.Package
+}
+
+// setSubsCatalogSource sets the Spec.source and Spec.sourceNamespace fields of a
+// Subscription, and returns the subscription
+func setSubsCatalogSource(subscription *olm.Subscription, csName string, csNamespace string) *olm.Subscription {
+	subscription.Spec.CatalogSource = csName
+	subscription.Spec.CatalogSourceNamespace = csNamespace
+	return subscription
+}
+
+func helperNewCatalogSourceConfig(namespace, name string) *v2.CatalogSourceConfig {
```

Comment:
This function seems named poorly. We should, in general, avoid a noun like `helper` when we are just describing a function instead of a type. I would buy that a `helper` would implement a method like `GenerateCatalogSourceConfig` and be called as `helper.GenerateCatalogSourceConfig()`.  But since there is no object, we should just call it something like `func GenerateCatalogSourceConfig()`.


Also you created this `cscInfo` type which has the fields you use as function arguments, but you don't use it here.